src/software/SfM/main_openMVG2MVE2.cpp

```diff
@@ -0,0 +1,299 @@
+        continue;
+
+      // Create current view subdirectory 'view_xxxx.mve'
+      std::ostringstream padding;
+      padding << std::setw(4) << std::setfill('0') << map_viewIdToContiguous[view->id_view];
+      sOutViewIteratorDirectory = stlplus::folder_append_separator(sOutViewsDirectory) + "view_" + padding.str() + ".mve";
+      stlplus::folder_create(sOutViewIteratorDirectory);
+      Intrinsics::const_iterator iterIntrinsic = sfm_data.GetIntrinsics().find(view->id_intrinsic);
+
+      // We have a valid view with a corresponding camera & pose
+      const std::string srcImage = stlplus::create_filespec(sfm_data.s_root_path, view->s_Img_path);
+      std::ostringstream os;
+      os << std::setw(5) << std::setfill('0') << map_viewIdToContiguous[view->id_view];
+      std::string dstImage = stlplus::create_filespec(
+        stlplus::folder_append_separator(sOutViewIteratorDirectory), "undistorted","png");
+
+      const IntrinsicBase * cam = iterIntrinsic->second.get();
+      if (map_viewIdToContiguous[view->id_view] == 0)
+        w_h_image_size = std::make_pair(cam->w(), cam->h());
+      else
+      {
+        // check that there is no image sizing change -- do we need to enforce this for MVE?
+        if (cam->w() != w_h_image_size.first ||
+            cam->h() != w_h_image_size.second)
+        {
+          std::cerr << "CMPMVS support only image having the same image size";
+          return false;
+        }
+      }
+      if (cam->have_disto())
+      {
+        // Undistort and save the image
+        ReadImage(srcImage.c_str(), &image);
+        UndistortImage(image, cam, image_ud, BLACK);
+        WriteImage(dstImage.c_str(), image_ud);
+      }
+      else // (no distortion)
+      {
+        // If extensions match, copy the PNG image
+        if (stlplus::extension_part(srcImage) == "PNG" ||
+          stlplus::extension_part(srcImage) == "png")
+        {
+          stlplus::file_copy(srcImage, dstImage);
+        }
+        else
+        {
+          ReadImage( srcImage.c_str(), &image);
+          WriteImage( dstImage.c_str(), image);
+        }
+      }
+
+      // Prepare to write an MVE 'meta.ini' file for the current view
+      const Pose3 pose = sfm_data.GetPoseOrDie(view);
+      Mat34 P = cam->get_projective_equivalent(pose);
+
+      for ( int i = 1; i < 3 ; ++i)
+        for ( int j = 0; j < 4; ++j)
+          P(i, j) *= -1.;
+
+      Mat3 R, K;
+      Vec3 t;
+      KRt_From_P( P, &K, &R, &t);
+
+      // Output translation via optical center vector for given pose
+      const Vec3 optical_center = R.transpose() * t;
+      // Pixel aspect = pixel width divided by the pixel height
+      const float pixelAspect = cam->w()/cam->h();
+      // Focal length and principal point are embedded into the calibration matrix K:
+      // focal_length = K(0,0)
+      // const std::vector<double> pp = {_K(0,2), _K(1,2)}
+      // their values are normalized (0..1)
+      const float flen = cam->h()/K(0, 0);
```

Comment:
According https://github.com/simonfuhrmann/mve/commit/12ca8538bd735b28b3a4da9e22da31b09cea9191 flen must be equal to cam->focal() / std::max(cam->h(), cam->w() );